docs/non-normative/how-to-write-conventions/resource-and-entities.md

```diff
@@ -0,0 +1,301 @@
+container-based entities are not enough, then new entities should be
+defined.
+
+### When to define an "is-a" relationship vs. extending descriptive attributes?
+
+There are two key rules:
+
+- Default to introducing separate entities with a clear "is a" (or similar) relationship
+- Extend an entity with new descriptive attributes if and only if the following is true:
+  - The extending entity cannot be associated with any telemetry by itself.
+    *For example, when adding a `windows.process` entity, but no new
+    signals (metrics, logs, spans, etc.) would be added to the entity.*
+  - The extending entity doesn’t have any other entities that logically can
+    only be associated with it.
+    *Note: this only applies to entities as signals with relationships.*
+
+This avoids the complexities of subtyping and ambiguous attribute usage.
+
+### How to define identifying attributes?
+
+The identifying attributes should be minimally sufficient to
+identify an entity within the context of how that entity is
+discovered. For example, when discovering kubernetes entities
+like `k8s.pod`, `k8s.deployment`, the identifying attributes
+should be sufficient to identify these entities within the
+scope of a kubernetes cluster (or more specifically, the
+kubernetes API server where the entities are discovered).
+
+Commonly, a number of attributes of an entity are readily
+available for the telemetry producer to compose an identity. Of
+the available attributes the entity ID should include the
+minimal set of attributes that is sufficient for uniquely identifying that entity. For example a process on a host can be
+uniquely identified by (`process.pid`,`process.creation.time`)
+attributes. Adding for example `process.executable.name`
+attribute to the identity is unnecessary and violates the
+rule of having a [minimally sufficient ID](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.45.0/specification/entities/data-model.md#minimally-sufficient-identity).
+
+#### Multi-Observer Guidance
+
+When choosing Identifying attributes, care should be taken to ensure that
+multiple observers will find the same identifying attribute for the same
+entity. Generally, entities may be discovered both within OpenTelemetry
+SDKs and the Collector and should leverage identifying attributes that
+will be the same between these signal providers.
+
+For example, `service.instance.id` can be problematic to detect from outside
+an SDK and inside an SDK consistently. Generally, this can only be
+achieved if some outside source injects a `service.instance.id` value into
+the SDK that is externally visible. An alternative is to have the SDK
+provide a relationship between the `service.instance.id` and another entity
+that is visible externally. Care should be taken when modelling Entities
+to avoid this problem where possible.
+
+The choice of `service.instance.id` should be an exception, not the rule, for
```

Comment:
>he choice of `service.instance.id` should be an exception, not the rule

I am confused. Is it necessary to say this? How can it be a rule for any other entity that is not a "Service"? Is it implied anywhere that it could be the rule?

Reply:
So the design decision for service exists because there is no central source of truth for what is a service. That should be the exception.  Other entities (e.g. k8s, cloud, etc.) should have some "holder" of entity information we can query/inspect that is stable across observers.

Reply:
Oh, I see. You are saying, the "Service" entity is special, don't try to design other entities the same way.